IPs/RSKIP326.md

```diff
@@ -1,56 +1,100 @@
+
+At the moment the `add_signature` event is being emitted before actually adding the signature.
 ## Motivation
 
-These changes will improve the user experience of getting event information from the blockchain in a human readable/friendly format. 
+### release_request_received format
+
+These changes will improve the user experience of getting event information from the blockchain in a human readable/friendly format.
+
+### New pegout_confirmed event
+
+At the moment clients don't have an easy way to know if a pegout has enough confirmations and is already waiting for signatures. With this new event, clients will be able know, probably live, when a pegout is already confirmed and waiting for signatures.
+
+An example of this is the 2wp-app/2wp-api. This tool informs users on the status of their pegout. Right now, to know if a pegout has been confirmed and is waiting for signature this tool makes intricate logic.
+
+### add_signature event logging order
+
+In some cases a signature will not be added to a transaction, so there would not be a need to actually emit the `add_signature` event.
 
 ## Specification
 
-### Current signature:
+### release_request_received changes
+
+#### Current signature:
 
 ```
 release_request_received(address indexed sender, bytes btcDestinationAddress, uint256 amount)
 ```
 
 The btcDestinationAddress field is currently in the bytes format.
 
-### Proposed signature:
+#### Proposed signature:
 
 ```
 release_request_received(address indexed sender, string btcDestinationAddress, uint256 amount)
 ```
 
 The btcDestinationAddress field will be updated to a user friendly string format.
 
-#### Data
+##### Data
 
 - sender: the RSK address of the peg-out requester.
 - btcDestinationAddress: the derived BTC address of said RSK address, where the BTC will be transferred to.
 - amount: amount in weis the user sent to the Bridge (this doesn't reflect how many BTC will be received since transfer fees will be deducted from that amount).
 
 If `RSKIP326` is activated, the event will log the `btcDestinationAddress` field in a user friendly string format.
 
+### pegout_confirmed event signature and description
+
+This event will exist after the batch pegout consensus changes (HOP, RSKIP271).
+
+```
+pegout_confirmed(bytes32 indexed btcTxHash, uint pegoutCreationBlockNumber)
+```
+
+- `btcTxHash`: the hash of the Bitcoin transaction without signatures that was just confirmed
+- `pegoutCreationBlockNumber`: the block where the pegout transaction was created, this is just informational value and totally optional
+
+### add_signature event logging order
+
+At the end of the `BridgeSupport::addSignature` method, we can see that is calling `eventLogger.logAddSignature` before `BridgeSupport::processSigning`.
+So, I propose to remove the `eventLogger.logAddSignature` call from `BridgeSupport::addSignature` and use it inside `BridgeSupport::processSigning` when the signature is actually added to the btcTx.
```

Comment:
I wouldn't go into code details.  You could just explain the general idea, only emmit add_signature event when the tx is actually signed and not when it tries to sign it